--- _tmp/openssh.rb	2024-11-13 20:56:30
+++ openssh.rb	2024-11-13 20:56:51
@@ -1,5 +1,5 @@
 class Openssh < Formula
-  desc "OpenBSD freely-licensed SSH connectivity tools"
+  desc "Farcloser: OpenBSD freely-licensed SSH connectivity tools"
   homepage "https://www.openssh.com/"
   url "https://cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-9.9p1.tar.gz"
   mirror "https://cloudflare.cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-9.9p1.tar.gz"
@@ -12,27 +12,15 @@
     regex(/href=.*?openssh[._-]v?(\d+(?:\.\d+)+(?:p\d+)?)\.t/i)
   end
 
-  bottle do
-    rebuild 1
-    sha256 arm64_sequoia: "d7941094cacc59b5c2ab1d64343751205ef31b7f9517b19bb2a128d6c3583226"
-    sha256 arm64_sonoma:  "6305fa9baff789a76925f68d983a7b402f2db8b2652baab7775a97477975ff53"
-    sha256 arm64_ventura: "f8245cffebafb4f31939f333eb903fcf7db285bcf6c0257e84644384602d827a"
-    sha256 sonoma:        "ab855f72ee7b1443d30256c6098c7f8593a6779c7ae6e9e2e12f6082c7406920"
-    sha256 ventura:       "6c763c4b4f845df7bb4ee967a85108835fc6382920b6461c3487546f4fab60a7"
-    sha256 x86_64_linux:  "84eb498f82816692da68f55d9bbc02bf8d7fb575f40c7efa008b6e9ac0738079"
-  end
-
   # Please don't resubmit the keychain patch option. It will never be accepted.
   # https://archive.is/hSB6d#10%25
 
   depends_on "pkg-config" => :build
   depends_on "ldns"
   depends_on "libfido2"
-  depends_on "openssl@3"
 
   uses_from_macos "mandoc" => :build
   uses_from_macos "lsof" => :test
-  uses_from_macos "krb5"
   uses_from_macos "libedit"
   uses_from_macos "libxcrypt"
   uses_from_macos "zlib"
@@ -81,9 +69,8 @@
       --sysconfdir=#{etc}/ssh
       --with-ldns
       --with-libedit
-      --with-kerberos5
       --with-pam
-      --with-ssl-dir=#{Formula["openssl@3"].opt_prefix}
+      --without-openssl
       --with-security-key-builtin
     ]
 
@@ -92,6 +79,103 @@
     system "./configure", *args
     system "make"
     ENV.deparallelize
+    rm "ssh_config"
+    touch "ssh_config"
+    inreplace "ssh_config", "", "
+Host *
+  # TODO: challenge these
+  Compression no
+  ForwardAgent no
+  Tunnel no
+
+  # Yubikey and macos integ
+  IgnoreUnknown UseKeychain
+  UseKeychain yes
+  AddKeysToAgent yes
+  IdentityFile ~/.ssh/id_ed25519_sk
+
+  AddressFamily any
+  BatchMode no
+  CheckHostIP yes
+  ConnectionAttempts 1
+  ConnectTimeout 20
+  EscapeChar ~
+  FingerprintHash sha256
+  ForwardX11 no
+  ForwardX11Trusted no
+  GatewayPorts no
+  HashKnownHosts yes
+  HostbasedAuthentication no
+  KbdInteractiveAuthentication no
+  PasswordAuthentication no
+  PermitLocalCommand no
+  Port 22
+  PubkeyAuthentication yes
+  StrictHostKeyChecking ask
+  CASignatureAlgorithms sk-ssh-ed25519@openssh.com,ssh-ed25519
+  HostKeyAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+  PubkeyAcceptedAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+  KexAlgorithms curve25519-sha256,sntrup761x25519-sha512@openssh.com
+  Ciphers aes256-ctr
+  MACs hmac-sha2-512-etm@openssh.com
+  HostbasedAcceptedAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+"
+
+    rm "sshd_config"
+    touch "sshd_config"
+    inreplace "sshd_config", "", "
+AcceptEnv LANG LC_*
+AddressFamily any
+AllowAgentForwarding no
+AllowTcpForwarding no
+ClientAliveInterval 300
+ClientAliveCountMax 3
+Port 22
+RekeyLimit 1G 3600
+AuthorizedKeysFile	.ssh/authorized_keys
+KbdInteractiveAuthentication no
+ChallengeResponseAuthentication no
+GatewayPorts no
+HostbasedAuthentication no
+IgnoreRhosts yes
+IgnoreUserKnownHosts yes
+ListenAddress 0.0.0.0
+ListenAddress ::
+LoginGraceTime 30s
+LogLevel INFO
+MaxAuthTries 1
+MaxSessions 10
+MaxStartups 3:100:4
+PasswordAuthentication no
+PermitEmptyPasswords no
+PermitRootLogin no
+PermitTunnel no
+PermitUserEnvironment no
+PrintLastLog no
+PrintMotd no
+Protocol 2
+PubkeyAuthentication yes
+AuthorizedKeysCommand none
+AuthorizedKeysCommandUser nobody
+StrictModes yes
+SyslogFacility AUTH
+TCPKeepAlive no
+UseDNS no
+UsePAM yes
+X11Forwarding no
+X11UseLocalhost yes
+PermitTTY yes
+VersionAddendum Magnetar/1.0
+# Crypto
+CASignatureAlgorithms sk-ssh-ed25519@openssh.com,ssh-ed25519
+HostKeyAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+PubkeyAcceptedAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+KexAlgorithms curve25519-sha256,sntrup761x25519-sha512@openssh.com
+Ciphers aes256-ctr
+MACs hmac-sha2-512-etm@openssh.com
+HostbasedAcceptedAlgorithms ssh-ed25519-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,ssh-ed25519,sk-ssh-ed25519@openssh.com
+"
+
     system "make", "install"
 
     # This was removed by upstream with very little announcement and has
@@ -101,9 +185,6 @@
 
     buildpath.install resource("com.openssh.sshd.sb")
     (etc/"ssh").install "com.openssh.sshd.sb" => "org.openssh.sshd.sb"
-
-    # Don't hardcode Cellar paths in configuration files
-    inreplace etc/"ssh/sshd_config", prefix, opt_prefix
   end
 
   test do
